docs: update pull request template for improved clarity and formatting (#200)
ga4gh/cat-vrs: .github/pull_request_template.md

> ## Link to the corresponding Issue.

Reviewer: ```suggestion
## Link to the corresponding Issue
```

Reviewer: This is more of a style preference of not punctuating headings

> ## Summary of the Pull Request.

Reviewer 2: ```suggestion
## Summary of the Pull Request
```

> ## Summary of the Pull Request.

Reviewer: ```suggestion
## Summary of the Pull Request
```

> ## Pull Request checklist:

Reviewer 2: ```suggestion
## Pull Request checklist
```

> ## Pull Request checklist:

Reviewer: ```suggestion
## Pull Request checklist
```

Reviewer: Outdated

### Required
- [ ] Does the title of this Pull Request reference the corresponding Issue?
- [ ] Is the branch validating against pre-commit hooks? Run `pre-commit run --all-files` from the root directory.
- [ ] Is the branch passing tests? Run `pytest tests/` from the root directory.

> 

Reviewer 2: I wonder if we should phrase as a confirmation rather than question?

```suggestion
- [ ] The title of this Pull Request accurately reflects the scope and content of the linked Issue.
- [ ] The branch passes all pre-commit hooks (Run `pre-commit run --all-files` from the root directory).
- [ ] The branch passes all tests (Run `pytest tests` from the root directory).
```

> ### If the schema or examples were contributed to:

Reviewer 2: ```suggestion
### Required if the schema or examples were contributed to
```

- [ ] Were the schema `def/` and `json/` files recompiled and committed? Run `cd schema; make all` from the root directory.
- [ ] Have tests been created or updated?
- [ ] Have schema changes been documented (update existing files or create new ones in `docs/source/`)?

> - [ ] If new `.rst` files were created, have they been registered in the documentation structure?

Reviewer: Hmm... this is the only piece that gives me pause. I worry that this statement may be a bit ambiguous. The part "new `.rst` files were created" could refer to either (i) new def/ files created in the schema folder, or (ii) new pages in the docs/ folder. You only intend to refer to the former, right? 

What do you think about the inclusion of "schema definition"  to the item?

```suggestion
- [ ] If new schema definition `.rst` files were created, have they been registered in the documentation structure?
```

Reviewer: Outdated

> - [ ] Has documentation been regenerated and committed? Run `cd docs; make clean watch &` from the root directory to compile documentation.

Reviewer 2: Same as above: I wonder if we should phrase as a confirmation rather than question?

```suggestion
- [ ] The schema `def/` and `json/` files have been recompiled and committed (Run `cd schema; make all` from the root directory).
- [ ] Tests have been created or updated.
- [ ] Schema changes have been documented (existing files updated or new files created in `docs/source/`).
- [ ] Any new `.rst` files have been registered in the documentation structure.
- [ ] Documentation has been regenerated and committed (Run `cd docs; make clean watch &` from the root directory).
```

Reviewer: I like that idea of stating it as a confirmation instead of question!

> - [ ] Any new `.rst` files have been registered in the documentation structure.

Reviewer: ```suggestion
- [ ] Any new schema definition `.rst` files have been registered in the documentation structure.
```

Reviewer: Re-adding this suggestion post-Kori's comments being incorporated. I would like this to differentiate .rst files generated by the schema from readthedoc pages. 

For example, assuming that the [PR about one concept per page gets incorporated](https://github.com/ga4gh/cat-vrs/pull/195):
- [/schema/cat-vrs/def/CategoricalVariant.rst](https://github.com/ga4gh/cat-vrs/blob/191-documentation-refactor/schema/cat-vrs/def/CategoricalVariant.rst?plain=1), generated by the schema 
- [/docs/source/concepts/CategoricalVariant.rst](https://github.com/ga4gh/cat-vrs/blob/191-documentation-refactor/docs/source/concepts/CategoricalVariant.rst?plain=1), the readthedocs page for Categorical Variant

Author: @brendanreardon I agree, this distinction makes sense.
